src/massDistribution/Ferriere.cpp

```diff
@@ -0,0 +1,330 @@
+#include "crpropa/massDistribution/Ferriere.h"
+
+namespace crpropa {
+
+
+Vector3d Ferriere::CMZTrafo(const Vector3d &position) const {
+	
+	// set galactocentric coordinate system with the Sun at (-8.5,0.,0.) instead of (8.5, 0, 0) to be consistand with JF12 implementation
+	double x = -position.x;
+	double y = -position.y;
+	
+	double xC = -50*pc;		//offset
+	double yC = 50*pc;	
+	double sinTc = sin(70./180.*M_PI);	//Tc = 70 deg
+	double cosTc = cos(70./180.*M_PI);
+	
+	
+	Vector3d pos;
+	pos.x = (x - xC)*cosTc + (y - yC)*sinTc;
+	pos.y = -(x - xC)*sinTc + (y - yC)*cosTc;
+	pos.z = position.z;
+		
+	
+	return pos;
+}
+
+Vector3d Ferriere::DISKTrafo(const Vector3d &position) const { 
+
+	// set galactocentric coordinate system with the Sun at (-8.5,0.,0.) instead of (8.5, 0, 0) to be consistand with JF12 implementation
+	double x = -position.x;
+	double y = - position.y;
+	double z = position.z;
+	
+	double alphaD = 13.5/180.*M_PI;	// rotation arround x-axis
+	double sinAd = sin(alphaD);
+	double cosAd = cos(alphaD);
+	double betaD = 20./180.*M_PI; // rotation arround y'-axis
+	double sinBd = sin(betaD);
+	double cosBd = cos(betaD);
+	double TettaD = 48.5/180.*M_PI; // rotation arround x"-axis
+	double sinTd = sin(TettaD);
+	double cosTd = cos(TettaD);
+	
+	Vector3d pos;
+	
+
+
+	
+	pos.x = x*cosBd*cosTd - y*(sinAd*sinBd*cosTd -cosAd*sinTd)-z*(cosAd*sinBd*cosTd +sinAd*sinTd);
+	
+	pos.y =  -x*cosBd*sinTd;
+	pos.y += y*(sinAd*sinBd*sinTd +cosAd*cosTd);
+	pos.y += z*(cosAd*sinBd*sinTd -sinAd*cosTd);
+	
+	pos.z = x*sinBd;
+	pos.z += y*sinAd*cosBd;
+	pos.z += z*cosAd*cosBd;		
+	
+	return pos;
+}
+	
+	
+
+
+double Ferriere::getHIDensity(const Vector3d &position) const {
+	
+	double n = 0;
+	double R = sqrt(position.x*position.x+position.y*position.y);	
+	bool inner = R<3*kpc;
+	
+	if(inner == true) 
+	{
+		double nCMZ = 0;	//density center in m^-3
```

Comment:
avoid declaring this as empty first, but later directly write

double nCMZ = 8.8 ....

this makes the code more readable